src/monitoring_service/service.py

```diff
@@ -163,16 +163,41 @@ def _process_new_blocks(self, last_block: BlockNumber) -> None:
 
         # check triggered events
         # TODO: create a priority queue for this
```

Comment:
Not for this PR, but now that I see it: I don't think we want a priority queue here, but instead we want to write these into the db, so that the events are not lost on crash. We also have to extend our crash test to cover such a scenario.

Reply:
Yes, both this and the other queue needs to be persisted to the DB

Reply:
See #157